test/replcompletions.jl

```diff
@@ -494,6 +493,27 @@ c, r, res = test_scomplete(s)
         ENV["PATH"] = oldpath
     end
 
+    # Make sure completion results are unique in case things are in the env path twice.
+    let
+        # Create /tmp/repl-completion and /tmp/repl-completion-subdir/repl-completion
+        file0 = joinpath(tempdir(), "repl-completion")
+        dir = joinpath(tempdir(), "repl-completion-subdir")
+        mkdir(dir)
+        file1 = joinpath(dir, "repl-completion")
+        touch(file0)
+        touch(file1)
+
+        withenv("PATH" => string(tempdir(), ":", dir)) do
+            s = string("repl-completio")
+            c,r = test_scomplete(s)
+            @test [utf8("repl-completion")] == c
+            @test s[r] == "repl-completio"
+        end
+
+        rm(file0)
```

Comment:
this should be in a `try / finally` block to clean up the temporary files / directory if the an exception is thrown